axelor-sale/src/main/java/com/axelor/apps/sale/exception/SaleExceptionMessage.java

```diff
@@ -128,4 +128,7 @@ private SaleExceptionMessage() {}
 
   public static final String SALE_ORDER_CONFIRM_TOTAL_AMOUNT_ZERO = /*$$(*/
       "The total amount of quotation is zero. Do you want to proceed ?" /*)*/;
+
+  public static final String PRODUCT_DOES_NOT_RESPECT_DOMAIN_RESTRICTIONS = /*$$(*/
+      "The product provided doesn't respect the domain's restrictions: %s" /*)*/;
```

Comment:
Please change the message to "You can not choose this product for this order".
And don't forget to add translation.